src/reducers/computeRenderedRows.js

```diff
@@ -163,13 +174,11 @@ function calculateRenderedRowRange(state, scrollAnchor) {
     }
   }
 
-  state.firstRowIndex = firstViewportIdx;
```

Comment:
moved back to caller. 

@wcjordan, sometimes it's hard to read through the code and see which state gets changed since a lot of functions mutate state directly. It's safe in our usages since state is always cloned, but just returning the value and then assigning it on the parent looks cleaner in places like this. Thoughts?